src/main/java/duke/parser/CommandProcessor.java

```diff
@@ -0,0 +1,109 @@
+package duke.parser;
+
+import duke.exception.DukeException;
+import duke.data.TaskList;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Arrays;
+
+public class CommandProcessor {
+    private final TaskList tasks;
+    private static final String[] VALIDCOMMANDS = {"mark", "unmark", "list", "todo"
+            , "event", "deadline", "delete", "find"};
+
+    private static final Storage storage = new Storage();

```

Comment:
I think it's a bit strange for the storage object to be stored under the parser. Perhaps find a way to keep it somewhere like Duke!

Reply:
Thanks for your input!